presto-local-file/src/main/java/com/facebook/presto/localfile/LocalFileTables.java

```diff
@@ -0,0 +1,149 @@
+ */
+package com.facebook.presto.localfile;
+
+import com.facebook.presto.spi.ColumnMetadata;
+import com.facebook.presto.spi.SchemaTableName;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+
+import java.io.File;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.OptionalInt;
+import java.util.concurrent.TimeUnit;
+
+import static com.facebook.presto.localfile.LocalFileMetadata.PRESTO_LOGS_SCHEMA;
+import static com.facebook.presto.localfile.LocalFileMetadata.SERVER_ADDRESS_COLUMN;
+import static com.facebook.presto.localfile.LocalFileTables.HttpRequestLogTable.*;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.presto.spi.type.TimestampType.TIMESTAMP;
+import static com.facebook.presto.spi.type.VarcharType.VARCHAR;
+import static com.google.common.base.Preconditions.checkArgument;
+
+public class LocalFileTables
+{
+    private final Map<SchemaTableName, DataLocation> tableDataLocations;
+    private final Map<SchemaTableName, LocalFileTableHandle> tables;
+    private final Map<SchemaTableName, List<ColumnMetadata>> tableColumns;
+
+    private final LoadingCache<SchemaTableName, List<File>> cachedFiles;
+
+    @Inject
+    public LocalFileTables(LocalFileConnectorId connectorId, LocalFileConfig config)
+    {
+        ImmutableMap.Builder<SchemaTableName, DataLocation> dataLocationBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<SchemaTableName, LocalFileTableHandle> tablesBuilder = ImmutableMap.builder();
+        ImmutableMap.Builder<SchemaTableName, List<ColumnMetadata>> tableColumnsBuilder = ImmutableMap.builder();
+
+
+        String httpRequestLogLocation = config.getHttpRequestLogLocation();
+        if (httpRequestLogLocation != null) {
+            Optional<String> pattern = Optional.empty();
+            if (config.getHttpRequestLogFileNamePattern() != null) {
+                pattern = Optional.of(config.getHttpRequestLogFileNamePattern());
+            }
+
+            SchemaTableName table = getSchemaTableName();
+            DataLocation dataLocation = new DataLocation(httpRequestLogLocation, pattern);
+            LocalFileTableHandle tableHandle = new LocalFileTableHandle(connectorId, table, getTimestampColumn(), getServerAddressColumn());
+
+            tablesBuilder.put(table, tableHandle);
+            tableColumnsBuilder.put(table, HttpRequestLogTable.getColumns());
+            dataLocationBuilder.put(table, dataLocation);
+        }
+
+        tables = tablesBuilder.build();
+        tableColumns = tableColumnsBuilder.build();
+        tableDataLocations = dataLocationBuilder.build();
+
+        cachedFiles = CacheBuilder.newBuilder()
+                .expireAfterWrite(1, TimeUnit.SECONDS)
```

Comment:
@electrum can you look at this cache config?  I'm concerned we may run into the problem we have in the Hive cache where you get old results the first time you run the query.  In this case, the table may not be used for days